Inspector: Improve method for checking if a process is inspectable

matthewbjones opened the pull request:
Improves the logic for the Inspector app to verify that the given `pid` is indeed inspectable.

The commit c9e849a defines whether or not a process is inspectable, defaulting to `No`.

The commit 4a2dd5b previously attempted to check whether or not a process could be inspected, but this seems to always fail and exit even when a process is indeed inspectable.

Two changes in this PR:

1. The `InspectorServer` IPC is updated to expose a new `is_inspectable` call, since the `InspectorServer` keeps track of what is currently inspectable.
2. The `Inspector` app uses this new IPC to determine if the selected / specified process is inspectable. If it is not, and you chose that process via the GUI, then the GUI reappears allowing you to select a different process. If you specified the `pid` via the command-line, then the app exits.

Fixes #7456

AnandBaburajan commented:
Hi @matthewbjones! Thank you for improving this.

> The commit [4a2dd5b](https://github.com/SerenityOS/serenity/commit/4a2dd5bf660fe177f7fdb3662c2a5b9a960b32c7) previously attempted to check whether or not a process could be inspected, but this seems to always fail and exit even when a process is indeed inspectable.

I'm not sure if "this seems to always fail and exit even when a process is indeed inspectable" is true because I (and another reviewer) checked out the changes and it seemed fine. I was able to inspect the Userland games' processes, for example.

But on the master branch now, I'm not able to inspect any of the Userland games.

matthewbjones commented:
@AnandBaburajan The commit c9e849a changed it so processes are not inspectable by default, and an application has to opt into being inspected. When this change was made, the Inspector would not inspect running processes I had in Serenity. Then, I changed c9e849a back to being `Yes`, and tried inspecting things like Solitaire, Terminal, etc. and it still would not inspect it because it would not pass the check here https://github.com/SerenityOS/serenity/commit/4a2dd5bf660fe177f7fdb3662c2a5b9a960b32c7#diff-b8d490cc79c79eff62923f0ab9143d25bd4fa01da5173dd28d00a7c8c193cf8fR100

Once I implemented the RPC with WindowServer, and temporary have the default back to `Yes` I could inspect Terminal, games, etc.

AnandBaburajan commented:
> The commit [c9e849a](https://github.com/SerenityOS/serenity/commit/c9e849a968175790d9cdbe0fe8f2e25ecb4a5dff) changed it so processes are not inspectable by default, and an application has to opt into being inspected. 

Oh, I should have looked into it. Thanks a lot for clarifying!
